src/utils/HighlightUtils.ts

```diff
@@ -277,4 +278,42 @@ export class HighlightUtils {
     }
     return highlighted;
   }
+
+  static highlight(text: string, match: string, className: string): HTMLElement[] {
```

Comment:
Open to name suggestions for this method,

Reply:
I wonder if, instead of having to highlight + escape at the same step, we could use `.text` to set the text content of the element and then loop through its `TextNode`, splitting & wrapping them to highlight them.

I've never done this, so I don't know to what extent it's possible, but if it's possible, it feels like it would be the most fool-proof solution.

Although the input here is directly from the user and only runs on their client, I'm slightly concerned by the current implementation (and the previous one) where a user's input is added to a regexp, since it seems vulnerable to RegExp denial of service.

Reply:
I think maybe you could reuse `DomUtils.highlight` or `DomUtils.highlightElement` 🤔

Reply:
Those methods have a very similar signature, but return a string. They need to use html encode to prevent XSS, because when a string is returned, it's rendered using: `innerHtml = ...string...`

 Since there is html encoding, it would not solve the reported bug,